File: VB/dxPaintValue/MyPropertyEditor.vb
Imports System.ComponentModel
Imports System.Drawing
Imports System.Drawing.Design

Namespace dxPaintValue
    Public Class MyPropertyEditor
        Inherits UITypeEditor

        Public Overrides Function GetPaintValueSupported(ByVal context As ITypeDescriptorContext) As Boolean
            Return True
        End Function

        Public Overrides Sub PaintValue(ByVal e As PaintValueEventArgs)
            If e.Value Is Nothing Then
                Return
            End If
            Dim color As Color = System.Drawing.Color.FromName(TryCast(e.Value, String))
            If Not color.IsEmpty Then
                Using brush As New SolidBrush(color)
                    e.Graphics.FillRectangle(brush, e.Bounds)
                End Using
            End If
        End Sub
    End Class
End Namespace

File: CS/dxPaintValue/MyPropertyEditor.cs
using System.ComponentModel;
using System.Drawing;
using System.Drawing.Design;

namespace dxPaintValue {
    public class MyPropertyEditor : UITypeEditor {
        public override bool GetPaintValueSupported(ITypeDescriptorContext context) {
            return true;
        }

        public override void PaintValue(PaintValueEventArgs e) {
            if(e.Value == null)
                return;
            Color color = Color.FromName(e.Value as string);
            if(!color.IsEmpty)
                using(SolidBrush brush = new SolidBrush(color))
                    e.Graphics.FillRectangle(brush, e.Bounds);
        }
    }
}
